python-libraries/data-platform-catalogue/data_platform_catalogue/client.py

```diff
@@ -382,17 +395,36 @@ def create_or_update_schema(self, metadata: DataProductMetadata, database_fqn: s
         """
         raise NotImplementedError
 
-    def create_or_update_table(self, metadata: TableMetadata):  # type: ignore[override]
+    def create_or_update_table(self, metadata: TableMetadata, platform: str = "glue", version: int = 1):  # type: ignore[override]
         """
-        Define a table.
-        There can be many tables per data product.
-        columns are expected to be a list of dicts in the format
+        Define a table (a 'dataset' in DataHub parlance), a 'collection of data'
+        (https://datahubproject.io/docs/metadata-modeling/metadata-model#the-core-entities).
+
+        There can be many tables per Data Product.
+
+        Columns are expected to be a list of dicts in the format
             {"name": "column1", "type": "string", "description": "just an example"}
+
+        This method creates a schemaMetadata aspect object from the metadata object
+        (https://datahubproject.io/docs/generated/metamodel/entities/dataset/#schemametadata)
+        together with generating a unique reference name (URN) for the dataset used by DataHub.
+        These are then emitted to the rest.li api as a dataset creation/update event proposal.
+
+        If tags are present in the metadata object, a second request is made to update the dataset
+        with these tags as a separate aspect.
+
+        Args:
+            metadata (TableMetadata): metadata object.
+            platform (str, optional): DataHub data platform type. Defaults to "glue".
+            version (int, optional): Defaults to 1.
+
+        Returns:
+            dataset_urn: the dataset URN
         """
         dataset_schema_properties = SchemaMetadataClass(
             schemaName=metadata.name,
-            platform=make_data_platform_urn("glue"),
-            version=1,
+            platform=make_data_platform_urn(platform=platform),
+            version=version,
             hash="",
             platformSchema=OtherSchemaClass(rawSchema="__insert raw schema here__"),
```

Comment:
this field is still a placeholder so I think we should either remove it (if not needed) or populate it

Reply:
yes , was hoping to mention the same

Reply:
what is  hash="" - used for ?. Can we remove any parameters if its not useful

Reply:
`platformSchema` is not an optional field, and needs to be populated. Unfortunately I'm not sure _what_ it needs to be populated _with_.

Same for `hash`.

Reply:
hash is "the SHA1 hash of the schema content" and platformSchema is "The native schema in the dataset's platform." according to https://datahubproject.io/docs/generated/metamodel/entities/dataset/#schemametadata

Don't know what we would actually pass though, or if it matters that much. We already have the native types passed through for each field, so maybe these can both be left as empty strings?